lib/number.js

```diff
@@ -53,12 +58,31 @@ internals.Number = class extends Any {
 
     multiple(base) {
 
-        Hoek.assert(Hoek.isInteger(base), 'multiple must be an integer');
-        Hoek.assert(base > 0, 'multiple must be greater than 0');
+        const isRef = Ref.isRef(base);
+
+        return this._test('multiple', base, (value, state, options) => {
+
+            let divisor;
+            if (isRef) {
+                divisor = base(state.parent, options);
+                if (typeof divisor !== 'number') {
+                    return this.createError('number.ref', { ref: base.key }, state, options);
+                }
+
+                if (divisor === 0) {
+                    return this.createError('number.refZero', { ref: base.key }, state, options);
+                }
+            }
+            else {
+                divisor = base;
+                if ((typeof divisor !== 'number') || (divisor === 0)) {
+                    return this.createError('number.illegalMultiple', { ref: base.key }, state, options);
+                }
+            }
 
-        return this._test('multiple', base, function (value, state, options) {
+            const mult = Math.pow(10, Math.max(internals.precision(value), internals.precision(divisor)));
```

Comment:
Yes @Marsup it is a real life problem for existing use cases in financial application involving exchange rates that can have 2 to 5 decimal accuracy.

number.multiple() fails to validate if an entered currency value is exact multiple of an amount in another currency.

**Example:**
6711 % 67.11 results in 5.684341886080802e-14
3230 % 32.3 results in 2.8421709430404007e-13

_Although in both the above cases it should have been 0 but modulo operator is broken for decimal/float values. (Check [JSFiddle](https://jsfiddle.net/santanubiswas/fL1zeadp/))_

Similar real world use cases also exist in stock trading application.